packages/processors/src/strategy/common/baseRecipientStatusUpdated.handler.ts

```diff
@@ -0,0 +1,133 @@
+import StatusesBitmap from "statuses-bitmap";
+import { Address, getAddress } from "viem";
+
+import { Application, Changeset, Round } from "@grants-stack-indexer/repository";
+import { ChainId, ProcessorEvent } from "@grants-stack-indexer/shared";
+
+import {
+    ApplicationStatus,
+    IEventHandler,
+    ProcessorDependencies,
+    RoundNotFound,
+} from "../../internal.js";
+import { createStatusUpdate, isValidApplicationStatus } from "../helpers/index.js";
+
+type Dependencies = Pick<
+    ProcessorDependencies,
+    "logger" | "roundRepository" | "applicationRepository"
+>;
+
+type ApplicationUpdate = {
+    application: Application;
+    status: number;
+};
+
+/**
+ * BaseRecipientStatusUpdatedHandler: Processes 'RecipientStatusUpdated' events
+ *
+ * - Decodes a bitmap containing status updates for multiple applications
+ * - Validates each status is valid (between 1-3)
+ * - Creates changesets to update application statuses in bulk
+ * - Serves as a base class as all strategies share the same logic for this event
+ *
+ * @dev:
+ * - Strategy handlers that want to handle the RecipientStatusUpdated event should create an instance of this class corresponding to the event.
+ *
+ */
+export class BaseRecipientStatusUpdatedHandler
+    implements IEventHandler<"Strategy", "RecipientStatusUpdatedWithFullRow">
+{
+    private readonly bitmap: StatusesBitmap;
+
+    constructor(
+        readonly event: ProcessorEvent<"Strategy", "RecipientStatusUpdatedWithFullRow">,
+        private readonly chainId: ChainId,
+        private readonly dependencies: Dependencies,
+    ) {
+        this.bitmap = new StatusesBitmap(256n, 4n);
```

Comment:
values to build `StatusesBitmap` are always the same ?

Reply:
yes, i followed original implementation. are constants values